library/choose_disk.py

```diff
@@ -353,46 +353,69 @@ def expand_disks(lookup_disks, ceph_type="", module=None):
     return final_disks
 
 
-def disk_label(partition):
+def disk_label(partition, current_fsid, ceph_disk):
     '''
     Reports if a partition is containing some ceph structures
     '''
-    # 1) let's search for legacy metadata made by ceph-disk
-    stdout = subprocess.check_output(["blkid", "-s", "PARTLABEL", "-o", "value", "{}".format(partition)])
 
+    # 1) let's search for metadata coming from ceph-volume
+    try:
+        output_cmd = subprocess.Popen(["ceph-volume", "lvm", "list", "--format=json", partition],
+                                      stdout=subprocess.PIPE)
+        raw_json, _ = output_cmd.communicate()
+        try:
+            json_dict = json.loads(raw_json)
+        except (ValueError, TypeError) as e:
+            fatal("Cannot parse ceph-volume properly : {}".format(e))
+
+        # If the disk have some metadata, the json is filled with information
+        for key, value in json_dict.items():
+            # As we parse a partition it should only have a single entry unless that's an error
+            if (len(value) > 1):
+                fatal("Error while parsing ceph-volume")
+            if "tags" in value[0]:
+                if "ceph.cluster_fsid" in value[0]['tags']:
+                    if value[0]["tags"]["ceph.cluster_fsid"] != current_fsid:
+                        return "foreign : {}".format(value[0]["tags"]["ceph.cluster_fsid"])
```

Comment:
what does "foreign" mean?

Reply:
"foreign" is a common term used on RAID systems. It describes a disk that has an existing RAID configuration. Typically the controller will see that information when a new disk is inserted and will mark this drive has "foreign". It's caution mechanism helping the administrator that the disk is part of another RAID.

In our case, this means we have found a drive that has existing Ceph metadata but from a **different** cluster than the current one.

Reply:
Commit message of 3983331058042958d41aa375f8c50cdb24a5a1bc does give a complete description.